lib/src/install.rs

```diff
@@ -1589,12 +1627,35 @@ pub(crate) async fn install_to_filesystem(
     if !st.is_dir() {
         anyhow::bail!("Not a directory: {root_path}");
     }
+
+    let inspect = crate::mount::inspect_filesystem(&fsopts.root_path)?;
+
+    let alternative_root = fsopts.root_path.join("sysroot");
+    let root_path = match inspect.source.as_str() {
```

Comment:
I recently changed this in https://github.com/containers/composefs/pull/303

I think we should detect the *type* of the filesystem and not its source string.

(Actually also today, ostree records whether it was booted via composefs; there's a file we can read, but eh)